### PR TITLE
milmove trace id

### PR DESCRIPTION
**Pull request by pjdufour-truss**
## Description

This PR creates a unique trace id for every http requests and a related logger that is passed down to each handler.  With this approach you can quickly correlate an `error` message with the specific request and any other `info` messages that were printed as part of that request.

Additionally, I've add an improvement to the `ecs-service-logs` program, so you add custom filters as positional arguments, which allows us to filter by `milmove-trace-id` (or any other property).

## Reviewer Notes

- `ecs-service-logs show -c app-experimental -s experimental -n 4 "milmove_trace_id=foo-bar"`
## Setup

```sh
make build_tools
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `scripts/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers](./docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @ntwyman
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/163989072) for this change

## Screenshots

N.A.

### CODE REVIEW

File: cmd/milmove/main.go [hunk 1]
```diff
@@ -431,7 +353,7 @@ func serveFunction(cmd *cobra.Command, args []string) error {
 	useSecureCookie := !isDevOrTest
 	// Session management and authentication middleware
 	noSessionTimeout := v.GetBool(cli.NoSessionTimeoutFlag)
-	sessionCookieMiddleware := auth.SessionCookieMiddleware(logger, clientAuthSecretKey, noSessionTimeout, appnames, useSecureCookie)
+	//sessionCookieMiddleware := auth.SessionCookieMiddleware(logger, clientAuthSecretKey, noSessionTimeout, appnames, useSecureCookie)
```
Comment (chrisgilmerproj) [inline cmd/milmove/main.go]: What's going on with this?

File: cmd/ecs-service-logs/main.go [hunk 2]
```diff
@@ -573,6 +573,15 @@ func showFunction(cmd *cobra.Command, args []string) error {
...
 	}
 
+	if len(args) > 0 {
+		for _, arg := range args {
+			parts := strings.SplitN(arg, "=", 2)
+			if len(parts) == 2 {
+				filters[parts[0]] = parts[1]
+			}
+		}
+	}
```
Comment (chrisgilmerproj) [inline cmd/ecs-service-logs/main.go]: Is this just saying that any additional args can be passed in without needing a flag? That's pretty neat actually. Could you add a comment though to explain it? Took me a while to track down what was going on by looking at it.

File: cmd/milmove/main.go (hunk 1 above)
Comment (chrisgilmerproj) [reply cmd/milmove/main.go]: Looks like this is resolved.

File: cmd/ecs-service-logs/main.go (hunk 2 above)
Comment (chrisgilmerproj) [reply cmd/ecs-service-logs/main.go]: I believe this was put in a different PR.

**Comment by rdhariwal** [comment]
Looks awesome, the other day Matt K and I were trying to figure out a user issue and a tracing id would have been great to get a sense of what was happening to a request lifecycle. 👍

**Pull request merged**